client/checkout/woopay/direct-checkout/index.js

```diff
@@ -9,59 +9,119 @@ import { debounce } from 'lodash';
 import WooPayDirectCheckout from 'wcpay/checkout/woopay/direct-checkout/woopay-direct-checkout';
 import { shouldSkipWooPay } from 'wcpay/checkout/woopay/utils';
 
 let isThirdPartyCookieEnabled = false;
 
-window.addEventListener( 'load', async () => {
-	if (
-		! WooPayDirectCheckout.isWooPayDirectCheckoutEnabled() ||
-		shouldSkipWooPay()
-	) {
+/**
+ * Handle the WooPay direct checkout for the given checkout buttons.
+ *
+ * @param {HTMLElement[]} checkoutButtons An array of checkout button elements.
+ */
+const handleWooPayDirectCheckout = async ( checkoutButtons ) => {
+	if ( ! checkoutButtons ) {
 		return;
 	}
 
-	WooPayDirectCheckout.init();
-
-	isThirdPartyCookieEnabled = await WooPayDirectCheckout.isWooPayThirdPartyCookiesEnabled();
-	const checkoutElements = WooPayDirectCheckout.getCheckoutRedirectElements();
 	if ( isThirdPartyCookieEnabled ) {
 		if ( await WooPayDirectCheckout.isUserLoggedIn() ) {
 			WooPayDirectCheckout.maybePrefetchEncryptedSessionData();
-			WooPayDirectCheckout.redirectToWooPay( checkoutElements, true );
+			WooPayDirectCheckout.addRedirectToWooPayEventListener(
+				checkoutButtons,
+				true
+			);
 		}
 
 		return;
 	}
 
 	// Pass false to indicate we are not sure if the user is logged in or not.
-	WooPayDirectCheckout.redirectToWooPay( checkoutElements, false );
+	WooPayDirectCheckout.addRedirectToWooPayEventListener(
+		checkoutButtons,
+		false
+	);
+};
+
+/**
+ * Add an event listener to the mini cart checkout button.
+ */
+const addMiniCartEventListener = async () => {
+	const checkoutButton = WooPayDirectCheckout.getMiniCartProceedToCheckoutButton();
+	handleWooPayDirectCheckout( [ checkoutButton ] );
+};
+
+/**
+ * If the mini cart widget is available on the page, observe when the drawer element gets added to the DOM.
+ *
+ * As of today, no window events are triggered when the mini cart is opened or closed,
+ * nor there are attribute changes to the "open" button, so we have to rely on a MutationObserver
+ * attached to the `document.body`, which is where the mini cart drawer element is added.
+ */
+const maybeObserveMiniCart = () => {
+	// Check if the widget is available on the page.
+	if ( ! document.querySelector( '.wc-block-mini-cart' ) ) {
+		return;
+	}
+
+	// Create a MutationObserver to check when the mini cart drawer is added to the DOM.
+	const observer = new MutationObserver( ( mutations ) => {
+		for ( const mutation of mutations ) {
+			if ( mutation?.addedNodes?.length > 0 ) {
+				for ( const node of mutation.addedNodes ) {
+					// Check if the mini cart drawer parent selector was added to the DOM.
+					if (
+						node.nodeType === 1 &&
+						node.matches(
+							'.wc-block-components-drawer__screen-overlay'
+						)
+					) {
+						// Wait until the button is rendered and add the event listener to it.
+						waitForSelector(
+							WooPayDirectCheckout.redirectElements
+								.BLOCKS_MINI_CART_PROCEED_BUTTON,
+							addMiniCartEventListener
```

Comment:
Yes. It's intentional as I tried to keep `waitForSelector` as generic as possible in case it gets used somewhere else.

In `addMiniCartEventListener` we could go with the callback element parameter, but since we were using `WooPayDirectCheckout.getCheckoutButtonElements()` and `WooPayDirectCheckout.getClassicProceedToCheckoutButton()`, I decided to keep the same pattern with `WooPayDirectCheckout.getMiniCartProceedToCheckoutButton()`.

Reply:
That makes sense. Thanks for confirming.